tianshou/env/worker/subproc.py

```diff
@@ -33,8 +34,8 @@
 class ShArray:
     """Wrapper of multiprocessing Array."""
 
-    def __init__(self, dtype: np.generic, shape: tuple[int]) -> None:
-        self.arr = Array(_NP_TO_CT[dtype.type], int(np.prod(shape)))  # type: ignore
+    def __init__(self, dtype: np.generic, shape: tuple[int], ctx: BaseContext) -> None:
```

Comment:
that's a breaking change, is it ok to do that? Can it be done in a backwards compatible way?

Reply:
It's only used within SubprocEnvWorker and by the time an object is created, the mp context has been created, so it might be ok?

Reply:
Ok, I guess it's fine then. Pls add a docstring. In particular, it should be clear how the class is supposed to be used, and how the context is being created. Mentioning an example in the code is enough for that.

The class needs to remain public b/c instances of it can be returned, it seems. But the user should be informed when they might be interested in it (if ever at all)

Reply:
I think it's fine to even pass `None` and create the default context in that case. Then it's a non breaking change